Data migration for User#remote_local_preferences

lenary opened the pull request:
Replace "Both" with "Local or Remote".  Currently have evil "monkey slap" lambda in users/edit.html.haml (removed from helper so I could nuke app/helpers entirely).

lenary commented:
i'll get this as it was i who previously monkey-slapped

lenary committed:
User#remote_local_preference bitch-slapping (migration)

- Added migration for "Both" to "Local or Remote"
- Removed the only UserPresenter tests, as that functionality no longer exists
- Updated factories and most of the rest of the code
- Renamed a function in UserPresenter to make more sense

lenary commented:
Really not sure this is the best way to do this. I think from efficiency of queries and everything that perhaps we should have two boolean fields -> `pairs_locally` and `pairs_remotely` or something - which we then use a presenter to display using these three strings

The only issue this would throw up is that we need to make sure they have checked both, or perhaps they could uncheck both to take a hiatus from pairing for a while without deleting their profile.

razielgn commented:
I'm totally with @lenary on this, storing the same string over and over will take away a lot of space on the database.

elight commented:
+1 though data will accumulate slowly. Even so, it's wasteful.

On Sep 10, 2011, at 3:01 PM, Federico Ravasioreply@reply.github.com wrote:

> I'm totally with @lenary on this, storing the same string over and over will take away a lot of space on the database.
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/rubypair/rubypair/pull/31#issuecomment-2060682

lenary committed:
Split User#remote_local_preference into two boolean fields

Now use User#remote and User#local (in views you can use 
UserPresenter#pairing_preference)

lenary commented:
Refactored into bool fields tests pass and I think i'm happy with the code.

elight commented:
Am I having a brain fart here?  The migrate:user_prefs task isn't showing up in rake. Hrm....

lenary commented:
have you checked out the correct branch?

elight commented:
Yes thanks.....

On Sep 22, 2011, at 5:02 AM, Sam Elliottreply@reply.github.com wrote:

> have you checked out the correct branch?
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/rubypair/rubypair/pull/31#issuecomment-2165485

lenary commented:
Hmmm - if you're running `rake -T` - the task won't show up as it has no description. If it's just not working, I have no idea. You're running this with bundle exec right?
